website/projects/spoke/src/app/core/components/tracking-popup/tracking-popup.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, ElementRef, Inject } from '@angular/core';
+import { MAT_SNACK_BAR_DATA } from '@angular/material/snack-bar';
+import { PageState } from '../../state/page.state';
+
+@Component({
+  selector: 'spoke-tracking-popup',
+  templateUrl: './tracking-popup.component.html',
+  styleUrls: ['./tracking-popup.component.scss']
+})
+export class TrackingPopupComponent {
+
+  container: HTMLElement;
+
+  constructor(elementRef: ElementRef<HTMLElement>, readonly page: PageState, @Inject(MAT_SNACK_BAR_DATA) public data: any) {
+    this.container = elementRef.nativeElement;
+  }
+
+  dismiss() {
+    this.page.setHasClosedPopup(true);
+    this.data.preClose();
+  }
+
+  allow() {
+    this.page.setAllowTelemetry(true);
+    console.log('allow tracking');
```

Comment:
console.log's / debug stuff should be removed

functions need to always report their return type (all these functions should be xxx(): void if they return nothing).